K2Field.K2NE.ServiceBroker/ServiceObjects/ExcelServices/ExcelImportServiceSO.cs

```diff
@@ -0,0 +1,528 @@
+                        // note the rows include header row
+                        if (!AreAllColumnsEmpty(temprow))
+                            dt.Rows.Add(temprow);
+                    }
+                }
+
+                // Here remove header row
+                dt.Rows.RemoveAt(0);
+                return dt;
+            }
+            catch (IOException ex)
+            {
+                throw new IOException(ex.Message);
+            }
+        }
+
+        /// <summary>
+        ///  Get Value in Cell 
+        /// </summary>
+        /// <param name="spreadsheetdocument">SpreadSheet Document</param>
+        /// <param name="cell">Cell in SpreadSheet Document</param>
+        /// <returns>The value in cell</returns>
+        private static string GetValueOfCell(SpreadsheetDocument spreadsheetdocument, Cell cell)
+        {
+            // Get value in Cell
+            SharedStringTablePart sharedString = spreadsheetdocument.WorkbookPart.SharedStringTablePart;
+            if (cell.CellValue == null)
+            {
+                return string.Empty;
+            }
+
+            string cellValue = cell.CellValue.InnerText;
+
+            // The condition that the Cell DataType is SharedString
+            if (cell.DataType != null && cell.DataType.Value == CellValues.SharedString)
+            {
+                return sharedString.SharedStringTable.ChildElements[int.Parse(cellValue)].InnerText;
+            }
+            else
+            {
+                return cellValue;
+            }
+        }
+
+
+        /// <summary>
+        /// Given a letter-number cell name reference, parses the specified cell element to get the column name.
+        /// </summary>
+        /// <param name="cellReference">Address of the cell (ie. B2)</param>
+        /// <returns>Column Name (ie. B)</returns>
+        private static string GetColumnRefName(string cellReference)
+        {
+            // Create a regular expression to match the column name portion of the cell name.
+            Regex regex = new Regex("[A-Za-z]+");
+            Match match = regex.Match(cellReference);
+
+            return match.Value;
+        }
+
+        /// <summary>
+        /// Given just the column name (no row index), it will return the zero based column index.
+        /// Note: This method will only handle columns with a length of up to two (ie. A to Z and AA to ZZ). 
+        /// A length of three can be implemented when needed. But if three is believed to be needed, question your requirements strongly!
+        /// </summary>
+        /// <param name="columnName">Column Name (ie. A or AB)</param>
+        /// <returns>Zero based index if the conversion was successful; otherwise null</returns>
+        public static int? GetColumnIndexFromRefName(string columnName)
+        {
+            int? columnIndex = null;
+            
+            char[] colLetters = columnName.ToCharArray();
+
+            if (colLetters.Count() <= 2)
```

Comment:
What if it's more than 2?

Reply:
This Service Object works only with 1 or 2 letters cells. It's quite a lot in Excel document.

Reply:
Done.